Docs for `kedro viz build` (#1710)
kedro-org/kedro-viz: docs/source/share_kedro_viz.md

When your project is ready, navigate to the root directory of the project.

## Publish and share Kedro-Viz automatically

> You can automate the process of publishing and sharing your Kedro-Viz. This section describes the steps to automate your Kedro-Viz deployment on AWS, which is the only cloud provider supported for automation at present. Integration with other cloud providers, namely Azure and GCP, will be added soon. [In the absence of automated publish and share for other platforms, there is a manual, platform-agnostic publish and share process described below](#platform-agnostic-sharing-with-kedro-viz). You can use the manual process for sharing on static website hosts like GitHub pages, and cloud providers like Azure and GCP.

Reviewer: ```suggestion
You can automate the process of publishing and sharing your Kedro-Viz. This section describes the steps for AWS, which is the only cloud provider supported for automation at present. Integration with other cloud providers, namely Azure and GCP, will be added soon. [In the absence of automated publish and share for other platforms, there is a manual, platform-agnostic publish and share process described below](#platform-agnostic-sharing-with-kedro-viz). You can use the manual process for sharing on static website hosts like GitHub pages, and cloud providers like Azure and GCP.
```

When your project is ready, navigate to the root directory of the project.

## Publish and share Kedro-Viz automatically

You can automate the process of publishing and sharing your Kedro-Viz. Currently we support AWS, but other cloud providers, namely Azure and GCP, are in pipeline. [There is a manual, platform-agnostic publish and share process described below](#platform-agnostic-sharing-with-kedro-viz) for static website hosts like GitHub pages, and cloud providers like Azure and GCP.

> This section describes the steps to follow to automate your Kedro-Viz deployment to AWS.

Reviewer 2: ```suggestion
This section describes the steps to automate your Kedro-Viz deployment on AWS.
```

> This section describes the steps to follow to automate your Kedro-Viz deployment to AWS.

Reviewer 3: I think what especially confused me is that Azure and GCP are mentioned both in the first and last sentence, making it a bit unclear if I should go for the automated or manual process when using a cloud provider.

> This section describes the steps to follow to automate your Kedro-Viz deployment to AWS.

Reviewer 3: It's much clearer now, thanks @stichbury!

> ### Update and install the dependencies

Reviewer: It would -- I don't think that was my commit that removed it (apologies if so). If you removed it @rashidakanchwala could you confirm if we should put it back?

When your project is ready, navigate to the root directory of the project.

## Publish and share Kedro-Viz automatically

> You can automate the process of publishing and sharing your Kedro-Viz. This section describes the steps for AWS, which is the only cloud provider supported for automation at present. Integration with other cloud providers, namely Azure and GCP, will be added soon. In the absence of automated publish and share for other platforms, there is a [manual, platform-agnostic publish and share process](#platform-agnostic-sharing-with-kedro-viz) described below. You can use the manual process for sharing on static website hosts like GitHub pages, and cloud providers like Azure and GCP.

Reviewer 4: This paragraph is a bit dense. Maybe it could be said "there are two ways to publish and share your Kedro-Viz: 1. the automatic way (only AWS is supported at present) and 2. the manual way (you can use any static hosting or object storage of your choosing)"

> You can automate the process of publishing and sharing your Kedro-Viz. This section describes the steps for AWS, which is the only cloud provider supported for automation at present. Integration with other cloud providers, namely Azure and GCP, will be added soon. In the absence of automated publish and share for other platforms, there is a [manual, platform-agnostic publish and share process](#platform-agnostic-sharing-with-kedro-viz) described below. You can use the manual process for sharing on static website hosts like GitHub pages, and cloud providers like Azure and GCP.

Reviewer 4: Also: "share your Kedro-Viz" should be replaced by "share your Kedro-Viz project", right? I recall there was a discussion around terminology a while ago cc @stichbury

> You can automate the process of publishing and sharing your Kedro-Viz. This section describes the steps for AWS, which is the only cloud provider supported for automation at present. Integration with other cloud providers, namely Azure and GCP, will be added soon. In the absence of automated publish and share for other platforms, there is a [manual, platform-agnostic publish and share process](#platform-agnostic-sharing-with-kedro-viz) described below. You can use the manual process for sharing on static website hosts like GitHub pages, and cloud providers like Azure and GCP.

Reviewer: I think it's a valid comment but equally is fine as it is (partly because I wrote it) so happy for Rashida to merge as is or make some further changes, depends on urgency to get this release out.

### Update and install the dependencies

> Kedro-Viz requires specific minimum versions of `fsspec[s3]`, and `kedro` to publish your project. Ensure you have these correct versions by updating the `requirements.txt` file in the `src` folder of the Kedro project to the following:

Reviewer 4: "In the `src` folder"

The `requirements.txt` file doesn't live there anymore, better to just say "the `requirements.txt` file"

> Kedro-Viz requires specific minimum versions of `fsspec[s3]`, and `kedro` to publish your project. Ensure you have these correct versions by updating the `requirements.txt` file in the `src` folder of the Kedro project to the following:

Reviewer: ```suggestion
Kedro-Viz requires specific minimum versions of `fsspec[s3]`, and `kedro` to publish your project. Ensure you have these correct versions by updating the `requirements.txt` file of the Kedro project to add the following:
```

When your project is ready, navigate to the root directory of the project.

## Platform-Agnostic sharing with Kedro-viz 

> Starting from Kedro-Viz version 7.1.0, we introduced the `kedro viz build` command. When a developer runs this command from the command line interface (CLI), it will create a `build` folder within their Kedro project. This build folder contains the entire static Kedro-Viz app package, which users can manually publish to any hosting platform of their choice.

Reviewer 2: [Nit] I am not sure of the tense but should it be - Starting from Kedro-Viz version 7.1.0, we **introduce** the `kedro viz build` command ?

Reviewer 2: When a developer runs this command from the command line interface (CLI) -> When a **user** runs this command from the command line interface (CLI) ?

Reviewer 2: This build folder contains the entire static Kedro-Viz app package which users can manually publish to any hosting platform of their choice. -> **The** build folder contains **a static Kedro-Viz app package, which can be used as a source to publish on any static website hosting platforms.**

Reviewer: I'm resolving this as I edited this section so no longer applicable

Some suggested platforms that have been explored with `kedro viz build` include:
* For static website hosting platforms like GitHub Pages:
    * Follow the steps linked [here](https://docs.github.com/en/pages/quickstart) to create a Git repository that supports GitHub Pages.
    * Once that's done, simply push the contents of the `build` folder to this new repository.
    * Your site will be available at the following URL: `http://<username>.github.io`
* For cloud providers such as AWS, Azure, and GCP:

>     * After setting up your bucket and enabling static hosting, manually update the contents of the `build` folder in the bucket.

Reviewer 2: After setting up your bucket and enabling static hosting, manually update the contents of the `build` folder in the bucket. -> **After creating a bucket and configuring it for static website hosting, copy the contents of the `build` folder to the bucket manually**

    * Access the Kedro-viz app via the bucket endpoint.

For AWS (and in the future Azure and GCP) we also offer deploy orchestration, which automates the upload process for you. This is explained in the next section.

> 

Reviewer 2: we also offer deploy orchestration, which automates the upload process for you. -> **we offer deploy orchestration, which automates the upload process.**

## Publish and share Kedro-viz on AWS 

> While `kedro viz build` offers a platform-agnostic and more manual way of publishing and sharing your Kedro-Viz, you can also automate the process using `kedro viz deploy`, which is currently supported only for AWS. More cloud platform support will be coming soon. Below are the steps to follow when deploying to AWS.

Reviewer 2: and more manual way of publishing - **, manual publishing and sharing of Kedro-Viz**

Reviewer 2: which is currently supported only for AWS. -> which currently **supports** AWS.

Reviewer 2: More cloud platform support will be coming soon -> Can we use a bracket or Note and mention 
**Support for other cloud providers namely Azure, GCP are in pipeline**

You pay for storing objects in your S3 buckets. The amount you pay depends on your objects’ size, how long you stored the object during the month, and the storage class.

See the official [AWS documentation](https://aws.amazon.com/s3/pricing/?nc=sn&loc=4) for more information. 

## Platform-agnostic sharing with Kedro-Viz 

In Kedro-Viz version 7.1.0, we introduced the `kedro viz build` command that enables you to publish and share Kedro-Viz to any static website hosting platform. Running this command from the command line interface (CLI) creates a `build` folder within the Kedro project. The build folder contains a static Kedro-Viz app package, which can be used as a source to publish on any static website hosting platform.

### Static website hosting platforms such as GitHub Pages

Follow the steps [listed in the GitHub pages documentation](https://docs.github.com/en/pages/quickstart) to create a Git repository that supports GitHub Pages. On completion, push the contents of the `build` folder to this new repository. Your site will be available at the following URL: `http://<username>.github.io`

> ### Cloud providers such as Azure, and GCP

Reviewer 2: [Nit] Can we include AWS as well and then at the end say For AWS, we also offer automated deployment ?

Author: For instance, we recently had a situation where someone was deploying to AWS. However, they encountered an issue because they had CloudFront enabled and were using their own DNS. As a result, they were unable to use the direct deploy method.

It's too early to say but I think direct deploy might cause issues. `kedro build` way might be the more robust chosen approach.

After creating a bucket and configuring it for static website hosting, copy the contents of the `build` folder to the bucket manually. You can then access the bucket via its endpoint.

> For AWS, we offer deploy orchestration to [automate the upload process as described above](#publish-and-share-kedro-viz-automatically).

Reviewer 4: Can we not use the word "orchestration" here to avoid confusion? Maybe "an automatic process"

Reviewer: ```suggestion
For AWS, we offer [automated deployment as described above](#publish-and-share-kedro-viz-automatically).
```

You pay for storing objects in your S3 buckets. The amount you pay depends on your objects’ size, how long you stored the object during the month, and the storage class.

See the official [AWS documentation](https://aws.amazon.com/s3/pricing/?nc=sn&loc=4) for more information. 

## Platform-agnostic sharing with Kedro-Viz 

> In Kedro-Viz version 7.1.0, we introduced the `kedro viz build` command to enable you to publish and share Kedro-Viz to any static website hosting platform. Running this command from the command line interface (CLI) creates a `build` folder within the Kedro project, which contains a static Kedro-Viz app package ready for to publish.

Reviewer 2: command to enable you to publish and share Kedro-Viz to any static website hosting platform. -> command **that enables** you to publish and share Kedro-Viz **on** any static website hosting platform.

> In Kedro-Viz version 7.1.0, we introduced the `kedro viz build` command to enable you to publish and share Kedro-Viz to any static website hosting platform. Running this command from the command line interface (CLI) creates a `build` folder within the Kedro project, which contains a static Kedro-Viz app package ready for to publish.

Reviewer: ```suggestion
In Kedro-Viz version 7.1.0, we introduced the `kedro viz build` command that enables you to publish and share Kedro-Viz to any static website hosting platform. Running this command from the command line interface (CLI) creates a `build` folder within the Kedro project, which contains a static Kedro-Viz app package ready for to publish.
```
